net.c

```diff
@@ -559,8 +559,12 @@ deliver_to_host(struct qitem *it, struct mx_hostentry *host)
 	READ_REMOTE_CHECK("MAIL FROM", 2);
 
 	/* XXX send ESMTP ORCPT */
-	send_remote_command(fd, "RCPT TO:<%s>", it->addr);
-	READ_REMOTE_CHECK("RCPT TO", 2);
+	to_addr = strtok(it->addr, ",");
+	while (to_addr != NULL) {
+		send_remote_command(fd, "RCPT TO:<%s>", to_addr);
+		READ_REMOTE_CHECK("RCPT TO", 2);
```

Comment:
couldn't this leave it->addr in a modified state?

Reply:
Indeed, you're 100% correct.  I just didn't notice it in testing since I think this is the last time that it->addr is currently used in the code.  To make it safer for future uses though, I just added a strdup.  Thanks for catching that.

Reply:
wouldn't we reuse the it if there is a temporary delivery error?